src/biokbase/narrative/jobs/job.py

```diff
@@ -603,3 +578,19 @@ def _create_error_state(
             "created": 0,
             "updated": 0,
         }
+
+    def _verify_children(self, children: List["Job"]) -> None:
+        if not self.batch_job:
+            raise NotBatchException("Not a batch container job")
+        if children is None:
+            raise ValueError(
+                "Must supply children when setting children of batch job parent"
+            )
+
+        inst_child_ids = [job.job_id for job in children]
+        if sorted(inst_child_ids) != sorted(self.child_jobs):
+            raise ValueError("Child job id mismatch")
```

Comment:
Is `_verify_children` going to be called in a way that if the children don't match `child_jobs`, the `children` can be updated?

(this question may be answered by scrolling down... I will find out soon!)

Reply:
Yup, it is. 👍